pkg/api/pod/util.go

```diff
@@ -884,6 +885,12 @@ func dropDisabledPodStatusFields(podStatus, oldPodStatus *api.PodStatus, podSpec
 		podStatus = &api.PodStatus{}
 	}
 
+	if !utilfeature.DefaultFeatureGate.Enabled(features.PodLevelResources) && !inPlacePodVerticalScalingInUse(oldPodSpec) {
```

Comment:
I think you need to add a new in-use function for this check, otherwise you could set these fields without the feature gate by setting the container-level fields (which will always be set automatically) and then setting the pod-level fields.